compiler.ts

```diff
@@ -138,13 +139,30 @@ function codeGenExpr(expr: Expr<Type>, env: GlobalEnv): Array<string> {
     case "binop":
       const lhsStmts = codeGenValue(expr.left, env);
       const rhsStmts = codeGenValue(expr.right, env);
-      return [...lhsStmts, ...rhsStmts, codeGenBinOp(expr.op)]
+      if (expr.op == BinOp.Is || expr.op == BinOp.And || expr.op == BinOp.Or) {
+        return [...lhsStmts, ...rhsStmts, codeGenBinOp(expr.op)]
+      }
+      // return [...lhsStmts, ...rhsStmts, codeGenBinOp(expr.op)]
+      var callName = codeGenBinOp(expr.op);
```

Comment:
Also, could you implement this in `lower`, by just directly compiling to the correct `call`?

Reply:
For the first question, having such a condition checking if ops is `eq`, `and`, or `or` is because we only defined the binary operations that interact with bignums. For those that don't interact with bignums, we just keep it the same. Therefore, for ops like `eq`, `and`, and `or`, it stays the same; but, for other binary operations, we then call the corresponding functions to handle the operations with bignums.

Reply:
We cleaned the code up a little so that everything is wrapped in `codeGenBinOp` now.